src/main/resources/handlebars/typescript-angular/api.module.mustache

```diff
@@ -18,7 +18,7 @@ import { {{classname}} } from './{{importPath}}';
     {{/hasMore}}{{/apis}}{{/apiInfo}} ]
 })
 export class ApiModule {
-    public static forRoot(configurationFactory: () => Configuration): ModuleWithProviders {
+    public static forRoot(configurationFactory: () => Configuration): ModuleWithProviders<ApiModule> {
```

Comment:
I think that for consistency with V2 and to allow backwards compatibility with versions of Angular <10 this should be:

`public static forRoot(configurationFactory: () => Configuration): ModuleWithProviders{{#genericModuleWithProviders}}<ApiModule>{{/genericModuleWithProviders}} {`

Reply:
Thanks for the feedback!
I actually copied the method from https://github.com/swagger-api/swagger-codegen/pull/10464 which automatically sets genericModuleWithProviders if angular is >= 7.